src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusCollectionManager.cs

```diff
@@ -182,19 +182,23 @@ private ExportResult OnCollect(Batch<Metric> metrics)
 
         try
         {
-            foreach (var metric in metrics)
+            if (this.exporter.ScopeInfoEnabled)
             {
-                if (!PrometheusSerializer.CanWriteMetric(metric))
+                var scopes = new HashSet<string>();
```

Comment:
potentially re-use this instead of newing up for each scrape/export.

Reply:
I've moved the hashset to the class instead. To have better caching, we'd need it on the `Batch<>`, which doesn't really belong there since this is a Prometheus-only feature.